backend/internal/server/middelwares.go

```diff
@@ -0,0 +1,55 @@
+package server
+
+import (
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt"
+)
+
+type DecodedJWT struct {
+	Name string
+    jwt.StandardClaims
+}
+
+func ExtractJWTInformation() gin.HandlerFunc {
```

Comment:
I think you dont need to decode the token by yourself. gin-keycloak can get information tou need here :
[See this](https://github.com/tbaehler/gin-keycloak?tab=readme-ov-file#how-to-get-the-keycloak-claims-eg-sub-mail-name)

Reply:
If it works, dont forget to remove unused deps

Reply:
Oh nice ... so far not working ... maybe a bug in the lib code
https://github.com/tbaehler/gin-keycloak/blob/3f68b881eb27a74eee8667cd60ebcffea7cf55e4/pkg/ginkeycloak/gin_keycloak.go#L264

Reply:
ok it's working